src/test/java/com/gw2auth/oauth2/server/service/OAuth2ScopesTest.java

```diff
@@ -0,0 +1,57 @@
+package com.gw2auth.oauth2.server.service;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.junit.jupiter.api.Test;
+
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+public class OAuth2ScopesTest {
+
+    @Test
+    public void allShouldContainAllGw2ApiPermissions() {
+        final Set<OAuth2Scope> all = OAuth2Scopes.all();
+        assertTrue(Gw2ApiPermission.stream().map(Gw2ApiPermission::scope).allMatch(all::contains));
```

Comment:
Nit: Although it does not matter for this particular case, I'd generally avoid constructs like `allMatch(all::contains)` because it will be faster to create an intermediate collection first (and passing it to `Set::containsAll`) most of the time. I.e., use:

```suggestion
        assertTrue(all.containsAll(Gw2ApiPermission.stream().map(Gw2ApiPermission::scope).toList()));
```

Reply:
testcase was removed as the complexity of the scopes became irrelevant to test